src/ui/base-page.js

```diff
@@ -0,0 +1,259 @@
+                namespace: "onContextChangeRequested",
+                priority: "last"
+            }]
+        },
+        invokers: {
+            reloadUioMessages: {
+                funcName: "sjrk.storyTelling.base.page.reloadUioMessages",
+                args: [
+                    "{arguments}.0",
+                    "{uio}.prefsEditorLoader.messageLoader",
+                    "options.locale"
+                ]
+            }
+        },
+        modelListeners: {
+            uiLanguage: [{
+                funcName: "sjrk.storyTelling.base.page.renderAllUiTemplates",
+                args: ["{that}"],
+                namespace: "renderAllUiTemplates"
+            },
+            {
+                funcName: "fluid.set",
+                args: ["{uio}", "options.multilingualSettings.locale", "{change}.value"],
+                namespace: "updateUioLanguage"
+            }],
+            "*": {
+                func: "{cookieStore}.set",
+                args: [null, "{page}.model"],
+                excludeSource: "init",
+                namespace: "setCookie"
+            }
+        },
+        components: {
+            cookieStore: {
+                type: "fluid.prefs.cookieStore",
+                options: {
+                    gradeNames: ["fluid.dataSource.writable"],
+                    cookie: {
+                        name: "sjrk-st-settings",
+                        path: "/",
+                        expires: ""
+                    }
+                }
+            },
+            // handles text to speech requests globally for the whole site
+            storySpeaker: {
+                type: "fluid.textToSpeech",
+                createOnEvent: "{page}.events.onAllUiComponentsReady",
+                options: {
+                    model:{
+                        ttsText: null,
+                        utteranceOpts: {
+                            lang: "{page}.model.uiLanguage"
+                        }
+                    },
+                    listeners: {
+                        "{page}.events.onStoryListenToRequested": {
+                            func: "{that}.queueSpeech",
+                            args: ["{that}.model.ttsText", true],
+                            namespace: "queueSpeech"
+                        }
+                    }
+                }
+            },
+            // the storytelling tool "main" menu
+            menu: {
+                type: "sjrk.storyTelling.ui.menu",
+                container: ".sjrkc-st-menu"
+            },
+            uio: {
+                type: "fluid.uiOptions.prefsEditor.multilingualDemo",
+                container: ".flc-prefsEditor-separatedPanel",
+                options: {
+                    components: {
+                        prefsEditorLoader: {
+                            options: {
+                                components: {
+                                    prefsEditor: {
```

Comment:
This is nested inconveniently deep. Instead factor off a prefsEditorInBasePage grade containing these listeners and send the grade down via an options distribution